src/borg/crypto/key.py

```diff
@@ -673,12 +673,14 @@ def change_passphrase(self, passphrase=None):
 
     @classmethod
     def create(cls, repository, args):
-        passphrase = Passphrase.new(allow_empty=True)
         key = cls(repository)
         key.repository_id = repository.id
         key.init_from_random_data()
         key.init_ciphers()
         target = key.get_new_target(args)
+        if os.path.isfile(target):
+            raise ValueError('Key in "%s" already exists. Not creating repository. Please move or rename this key first.' % target)
```

Comment:
This will generate a traceback (which is not very useful for expected errors with known root cause).

You can use `Error` or a subclass of it to avoid the traceback, give a prettier error msg (but otherwise behave similarly).

Reply:
Also, `target` seems to be a repository instance, this is why tests are failing currently.

Reply:
guess you need to move this check to `KeyfileKey.save` - there it is already clear that we deal with some specific path and a local file (not a `RepoKey`).

Reply:
done.